packages/cli/src/controller/publish-controller.ts

```diff
@@ -78,10 +78,11 @@ export async function uploadToIpfs(
       throw new Error('Unable to parse project manifest');
     }
 
+    // the JSON object conversion must occur on manifest.deployment
     const deployment = await replaceFileReferences(reader.root, manifest.deployment, authToken, ipfs);
 
     // Use JSON.* to convert Map to Object
-    const deploymentStr = yaml.dump(JSON.parse(JSON.stringify(deployment)), {sortKeys: true, condenseFlow: true});
+    const deploymentStr = yaml.dump(deployment, {sortKeys: true, condenseFlow: true});
```

Comment:
Maybe we can move this to a function on a deployment? That way we can be consistent with the options.